### PR TITLE
Add ability to reset date selection

### PR DESCRIPTION
**Pull request by chrishanes**
This is a JS solution only as a normal form reset without JS will not clear radio button selections

### CODE REVIEW

File: spec/javascripts/clear_form_radio_buttons_spec.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    it('label should have `selected` class', function() {
+      expect($('#visit_slot_granted').parent('label')).toHaveClass('selected');
+    });
+  });
+
+  describe('should be unchecked', function() {
+    beforeEach(function() {
+      moj.Modules.clearRadioButtons.init();
+      spyOnEvent('.js-clearRadioButton', 'click');
```
Comment (StupidCodeFactory) [inline spec/javascripts/clear_form_radio_buttons_spec.js]: I'm not sure why you would spy on the the click event if it get triggered the line after?

Comment (chrishanes) [inline spec/javascripts/clear_form_radio_buttons_spec.js]: I forgot to add the test to test the click event

Comment (StupidCodeFactory) [inline spec/javascripts/clear_form_radio_buttons_spec.js]: I see, i don't really understand the value of that test. To me it seems you're that triggering an event triggers the event. You would want to spy on an event if it is triggered by something else like from a calling function.

Comment (chrishanes) [inline spec/javascripts/clear_form_radio_buttons_spec.js]: I was bored

Comment (StupidCodeFactory) [inline spec/javascripts/clear_form_radio_buttons_spec.js]: I think it's safe to remove the spy and the extra test

**Pull request merged**